lib/ai_moderation/spam_scanner.rb

```diff
@@ -394,22 +394,32 @@ def self.handle_spam(post, log)
             queue_for_review: true,
           ).perform
 
-        log.update!(reviewable: result.reviewable)
+        # Currently in core re-flagging something that is already flagged as spam
+        # is not supported, long term we may want to support this but in the meantime
+        # we should not be silencing/hiding if the PostActionCreator fails.
+        if result.success?
+          log.update!(reviewable: result.reviewable)
+
+          reason = I18n.t("discourse_ai.spam_detection.silence_reason", url: url)
+          silencer =
+            UserSilencer.new(
+              post.user,
+              flagging_user,
+              message: :too_many_spam_flags,
+              post_id: post.id,
+              reason: reason,
+              keep_posts: true,
+            )
+          silencer.silence
 
-        reason = I18n.t("discourse_ai.spam_detection.silence_reason", url: url)
-        silencer =
-          UserSilencer.new(
-            post.user,
-            flagging_user,
-            message: :too_many_spam_flags,
-            post_id: post.id,
-            reason: reason,
-            keep_posts: true,
+          # silencer will not hide tl1 posts, so we do this here
+          hide_post(post)
+        else
+          log.update!(
+            error:
+              "unable to flag post as spam, post action failed for post #{post.id} with error: '#{result.errors.full_messages.join(", ")}'",
```

Comment:
this will error out when errors are longer than 3000 :) we probably need to truncate.

Reply:
Yeah was wondering about that, I don't _think_ the errors will be that long, but will truncate just in case :)